### PR TITLE
UI DataTable: Add LinkList Column

### PR DESCRIPTION
**Pull request by tfamula**
Bildschirmfoto vom 2023-10-26 17-32-16

### CODE REVIEW

**Comment by tfamula** [comment]
It would be great if we could fasttrack this before Coding Completed tomorrow, like https://github.com/ILIAS-eLearning/ILIAS/pull/6459.

File: src/UI/examples/Table/Column/Link/base.php [hunk 1]
```diff
@@ -55,6 +55,7 @@ public function getTotalRowCount(
         }
     };
 
-    $table = $f->table()->data('Link Columns', $columns, $data_retrieval);
+    $table = $f->table()->data('Link Columns', $columns, $data_retrieval)
+        ->withRequest($DIC->http()->request());
```
Comment (tfamula) [inline src/UI/examples/Table/Column/Link/base.php:59]: This PR also includes a little fix for the the example of the Link Column

**Review by thibsy — CHANGES_REQUESTED** [review]
Hi @tfamula

Thanks a lot for your contribution to the UI components!

It is possible to fast-track this as well, like we did with #6459. So if you can provide the requested changes today, I will gladly merge this afterwards.

Please consider the following suggestions. You do not need to follow them, but please indicate shortly why you prefer to do otherwise:

- [x] `Column\LinkList::format()`: this method currently compares the given `$value` against the `Listing\Listing` interface. IMO this is too broad and would allow any kind of listing in the future (even though they might fail `Link\Standard` check like `Listing\Property` currently does). I believe we should stick with only `Listing\Ordered` and `Listing\Unordered` for the moment.

Please implement the following changes:

- [x] `LinkList`: please rename this interface to `LinkListing`, indicating its association with `Listing\*` components. Also rename the factory method accordingly.
- [x] `ColumnTest`: please also add a unit-test for invalid values passed into `LinkList::format()`. One where another component is passed, and, one where a listing with non-link values is passed.

Kind regards,
@Amstutz, @klees and @thibsy

**Comment by tfamula** [comment]
Hi @thibsy 
Thanks for the quick feedback! All done!

**Review by thibsy — APPROVED** [review]
Thx @tfamula, this LGTM now. I added the JF label so we can announce this as well.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 30 OCT 2023: We highly appreciate this suggestion and accept the PR for Trunk.

**Pull request merged**